content/en/blog/2020/new-deployment-model/index.md

```diff
@@ -0,0 +1,103 @@
+
+It is common to have all these personas work on the same clusters without any clear separation.  For example, there are multiple ways to deploy Istio according to the docs, which all started with mesh operator, platform owner and service owner sharing the single cluster first and then gradually expanding the mesh to multiple clusters or VMs.  None of these provided a clear separation between mesh operator and platform/service owner at the boundary of a cluster.  You may be thinking you could set up [Kubernetes RBAC rules](https://kubernetes.io/docs/reference/access-authn-authz/rbac/) and namespaces to control which personas can do what within the cluster, however, sometimes you need stronger separation between the istio control plane and the rest at the cluster level.  Introduce a new deployment model for Istio which enables mesh operators to install and manage mesh control plane on dedicated clusters, separated from the data plane clusters.  This new deployment model could easily enable Istio vendors to run Istio control plane for service mesh users while users can focus on their workloads and Istio resources without worrying about installing or managing the Istio control plane.
+
+
+## New Deployment model
+
+Let’s take a close look of what has been deployed in your cluster if you install Istio following the default install:
+* Deployment and Services:
+    * Istiod: Istio control plane component, which includes an XDS server, CA server and a webhook server.
+    * Gateway: the default Istio ingress and egress gateways.
+* Configs:
+    * CRDs: Various Istio CRDs such as Gateway, Virtual Services, Destination Rules etc.
+    * ConfigMaps: Istio mesh configurations and Injection templates.
+    * Webhook configurations: configurations for two Istio webhooks (validation webhook and sidecar injector webhook)
+    * Service account, role binding, etc: Istio security settings
+    * Secrets: stores key and certs and credentials related to service account or Istiod.
+
+Below diagram shows how resources are deployed in the cluster after following the [default install](https://istio.io/latest/docs/setup/install/istioctl/#install-istio-using-the-default-profile).
+
+{{< image
+    link="./default-install.jpeg"
+    caption="default deployment model for istio"
+>}}
+
+With istio 1.7, it is possible to run Istiod on a separate cluster like the diagram below.
+
+{{< image
+    link="./central-istiod-single-cluster.jpeg"
+    caption="New deployment model for one data plane cluster"
+    >}}
+
+In this setup, the `cluster1` should be set up first with `istiodRemote` and `base` components, then istiod on the control plane cluster is installed second, with its KUBECONFIG configured to `cluster1`. With this new deployment model, mesh operators can work purely on the control plane cluster when there is a need to upgrade or reconfig istio. Platform owners and service owners can instead work solely on cluster1 to focus on their istio configs or services.  
```

Comment:
I think this article needs to explain how a `control plane cluster` is different from a [primary cluster](https://preliminary.istio.io/latest/docs/reference/glossary/#primary-cluster). It fits the definition of a primary cluster, just one that has nothing else other than the control plane running in it. To avoid confusion, I think this needs to explain what's different in the implementation for this case. If there is a fundamental difference, then we should consider adding another term to the glossary for `control plane cluster: a primary cluster that runs a control plane only`, or something like that.

Reply:
Thanks @frankbu for the review!  I am thinking we may need a new terminology in this blog or in https://preliminary.istio.io/latest/docs/reference/glossary/#primary-cluster.

@linsun WDYT?

Reply:
I would not call this primary cluster for multicluster.  Primary cluster is a concept used within multicluster.  Here, we start with single cluster.  The control plane cluster is the cluster that runs one or more Istiod control planes.

Reply:
The problem is that it fits the current definition of a primary cluster "a primary cluster is a cluster with a control plane". We'll need to clarify that definition and this article needs to be clear how a "control plane cluster" is different. How, for example, do you achieve HA control planes? Where does all this fit into the [deployment guide](https://preliminary.istio.io/latest/docs/ops/deployment/deployment-models/)?

cc @smawson @nmittler

Reply:
@linsun @frankbu @smawson `primary cluster` is not specifically a multicluster concept, it's simply a cluster with a control plane. Adding another, similar term here will simply cause confusion.  I think before this change lands, we should first agree on a set up updated glossary terms.

@linsun this all goes back to a discussion that we had around central istiod. We need to break it down into specific concepts. To get the discussion rolling, I propose the following:

- Primary Cluster: a cluster with a control plane (no change from current definition).
- Config Cluster: a cluster that provides configuration for one or more Istio control planes.

WRT the use of `central istiod`, this seems to be in conflict with the existing term [Managed Control Plane](https://istio.io/latest/docs/reference/glossary/#managed-control-plane). I think we either need to abandon the use of `central istiod` or discuss its use and how it's different from `managed control plane` with the TOC.

Reply:
I'll run this through the env WG and TOC if needed.  This blog can be easily updated by changing central to managed if we prefer.  In the last week's env wg mtg, seems most folks prefer to use central.

Reply:
Sounds good, @linsun ... thanks!

The issue with `central` as I see it is that it's simply in conflict with `managed`, so it's less about individual opinions and more about how to reconcile the two.

What are your thoughts on using `primary cluster`?

Reply:
Primary cluster in multicluster refers to the control plane cluster in general, however in central/managed istiod env, since control plane no longer runs in user's clusters, primary cluster refers to primary data plane cluster.

WDYT @nmittler ?

Reply:
@linsun no need to run this through env WG. Our collective decision as I recall it was to let it be named what it is named with of course TOC oversight. My takeaway was the original  is already in place, and is such is sticky.

Reply:
I wouldn't use "central" because central brings to mind "single point of failure" which is not a good thing. It also brings up the question of how do you get control plane HA with this model if istiod is running in only one cluster?

As I said above, I also still think that this blog shouldn't be introducing new terminology, but instead introducing the example and referring to diagrams and terms that are added to the Deployment Models doc, not put them in the blog itself. The blog can be a very short intro and just point to the model it's implementing, instead of confusing readers by describing things that Istio docs don't mention.

Also, I don't think `primary cluster` can be used to mean `primary data plane cluster` in the managed/central istiod model, because primary -> control plane. None of the mesh clusters have a control plane so there are zero primary clusters in this model. All the clusters are `remote clusters`.

I think the fundamental difference between a `primary cluster` and a `control plane cluster` is that the a primary cluster is a "mesh cluster" with a control plane, while a control plane cluster is just a "plain K8s cluster" with a control plane.

I think this should all be described clearly in the deployment models doc, not in a blog.

Reply:
@frankbu I like your comment about not introducing new terminology and make this simple to focus on user cases.   I agree with your difference about `primary cluster` vs `control plane cluster` as well.

Are you suggesting we update deployment model first before releasing a blog?

Reply:
> I think the fundamental difference between a primary cluster and a control plane cluster is that the a primary cluster is a "mesh cluster" with a control plane, while a control plane cluster is just a "plain K8s cluster" with a control plane.

I strongly disagree. The intent for `primary cluster` is simply to suggest that it is a cluster where the control plane runs. It has no distinction of whether user workloads run there or not. It's just a cluster that runs a control plane. You can certainly have a dedicated primary cluster that runs no user workloads ... that's fine. Adding `control plane cluster` will be a huge problem for terminology ... it will simply be confused with `primary cluster` in every conversation. In fact, when we decided on the term `primary cluster`, `control plane cluster` was one of the candidate terms. We're better off not introducing a new term for this and simply refer to it as a `primary cluster` that runs outside of the mesh.

Reply:
The problem is that implementation wise, Istiod deployed in a primary cluster is different than Istiod deployed as an external control plane. It's not just the same instructions, deploy Istio and then either do or don't run any services next to it. I'm not dwelling on the distinction in my proposed doc updates. PTAL at #8015

Reply:
I still don't think that's a valid distinction. The configuration difference IIUC is to enable the control plane to use another cluster as a source for config. That is orthogonal to whether or not the control plane is external. Theoretically we could use an in-mesh cluster configured to use another cluster as its source of config.